app/src/ui/changes/filter-changes-list.tsx

```diff
@@ -1149,39 +1197,254 @@ export class FilterChangesList extends React.Component<
+    this.filterWarningButtonRef = buttonRef
+  }
+
+  private openFilterWarning = () => {
+    this.setState({ isFilesWarningOpen: true })
+  }
+
+  private closeFilterWarning = () => {
+    this.setState({ isFilesWarningOpen: false })
+  }
+
+  private renderFilterWarning() {
+    return (
+      <Popover
+        className="filter-popover"
+        ariaLabelledby="filter-warning-header"
+        anchor={this.filterWarningButtonRef}
+        anchorPosition={PopoverAnchorPosition.BottomRight}
+        decoration={PopoverDecoration.Balloon}
+        onMousedownOutside={this.closeFilterWarning}
+        onClickOutside={this.closeFilterWarning}
+      >
+        <div className="filter-popover-header">
+          <h3 id="filter-warning-header">Committing Hidden Files</h3>
+          <button
+            className="close"
+            onClick={this.closeFilterWarning}
+            aria-label="Close"
+          >
+            <Octicon symbol={octicons.x} />
+          </button>
+        </div>
+        <div className="filter-warning-text">
+          You have a filter applied. There are{' '}
+          <LinkButton onClick={this.showFilesToBeCommitted}>
+            hidden changes
+          </LinkButton>{' '}
+          that will be committed.
+        </div>
+      </Popover>
+    )
+  }
+
+  private openFilterOptions = () => {
+    if (!this.props.canFilterChanges) {
+      this.props.dispatcher.setCanFilterChanges(true)
+      return
+    }
+    this.setState({ isFilterOptionsOpen: true })
+  }
+
+  private closeFilterOptions = () => {
+    this.setState({ isFilterOptionsOpen: false })
+  }
+
+  private renderFilterOptions() {
+    const checkedFilesThatAreVisibleCount = [
+      ...this.state.filteredItems.values(),
+    ].filter(
+      f =>
+        this.props.workingDirectory
+          .findFileWithID(f.id)
+          ?.selection.getSelectionType() !== DiffSelectionType.None
+    ).length
+
+    return (
+      <Popover
+        className="filter-popover"
+        ariaLabelledby="filter-options-header"
+        anchor={this.filterOptionsButtonRef}
+        anchorPosition={PopoverAnchorPosition.BottomRight}
+        decoration={PopoverDecoration.Balloon}
+        onMousedownOutside={this.closeFilterOptions}
+        onClickOutside={this.closeFilterOptions}
+      >
+        <div className="filter-popover-header">
+          <h3 id="filter-options-header">Filter Options</h3>
+          <button
+            className="close"
+            onClick={this.closeFilterOptions}
+            aria-label="Close"
+          >
+            <Octicon symbol={octicons.x} />
+          </button>
+        </div>
+        <div className="filter-options">
+          <Checkbox
+            value={
+              this.state.filterToIncludedCommit
+                ? CheckboxValue.On
+                : CheckboxValue.Off
+            }
+            onChange={this.onFilterToIncludedInCommit}
+            label={`Being committed (${checkedFilesThatAreVisibleCount})`}
```

Comment:
This label of "Being committed" has been discussed a few time as to what is the clearer verbiage.. It could also be "Checked" if we wanted a more succinct visual/state linked description.

Reply:
Bah! Meant to comment on this too 🤦‍♀️ My 2 cents:

```suggestion
            label={`Selected changes only (${checkedFilesThatAreVisibleCount})`}
```

Reply:
> Selected

I don't want to use this because our selected state across the app is the "blue highlight" and in this scenario means that it is the file selected to be viewed in the diff NOT selected (or checked) to be committed. This in the semantics as well; screenreaders users where here "selected" and "checked" when navigating list items if the item is checked.